driver/src/price_finding/price_finder_interface.rs

```diff
@@ -36,20 +40,51 @@ impl FromStr for SolverType {
             "standard-solver" => Ok(SolverType::StandardSolver),
             "fallback-solver" => Ok(SolverType::FallbackSolver),
             "naive-solver" => Ok(SolverType::NaiveSolver),
+            "open-solver" => Ok(SolverType::OpenSolver),
             _ => Err(anyhow!("solver type does not exit")),
         }
     }
 }
 
 impl SolverType {
-    pub fn to_args(self) -> Vec<String> {
-        match self {
-            SolverType::StandardSolver => vec![],
-            SolverType::FallbackSolver => vec![String::from("--useExternalPrices")],
+    pub fn execute(
```

Comment:
Not important, as it's pretty internal but I find this method a but messy (>30 lines and some variables which are only necessary in some cases make this hard to follow).

Maybe we can have private helpers `execute_open_solver` and `execute_private_solver(use_external_prices)` and then only one match statement in this method that calls the helpers or panics.